### PR TITLE
197 fix styled expressions

### LINKED ISSUE #197: Interpolation in styled template literal doesn't support values or expressions
**Describe the bug**
When using the styled component API, expression interpolation only seems to work when using inline functions or a variable that points to a value.

**To Reproduce**
The following compiles successfully but produces broken CSS:

```
const Div = styled.div`
  border-radius: ${2 + 2}px;
`;
```

```
const br = 2 + 2;
const Div = styled.div`
  border-radius: ${br}px;
`;
```

The following fails to compile:

```
const Div = styled.div`
  border-radius: ${4}px;
`;
```

```
const Div = styled.div`
  border-radius: ${'4'}px;
`;
```

```
const Div = styled.div`
  border-radius: ${`${4}`}px;
`;
```

```
const getBr = () => 4;
const Div = styled.div`
  border-radius: ${getBr}px;
`;
```

These work correctly, however:

```
const br = 4;
const Div = styled.div`
  border-radius: ${br}px;
`;
```

```
const Div = styled.div`
  border-radius: ${() => 2 + 2}px;
`;
```

**Desktop (please complete the following information):**
 - OS: macOS
 - Browser: Chrome
 - Version: 0.3.2

### PR DESCRIPTION
**Pull request by madou**
Fixes #197

### CODE REVIEW

File: packages/ts-transform/src/utils/css-transform.tsx [hunk 1]
```diff
@@ -49,9 +49,9 @@ const parentOrphenedPseudos = plugin('parent-orphened-pseudos', () => {
 
 const extractStyleSheets = plugin<{ callback: (sheet: string) => void }>(
   'extract-style-sheets',
-  opts => {
-    return root => {
-      root.each(node => {
+  (opts) => {
+    return (root) => {
+      root.each((node) => {
```
Comment (danieldelcore) [inline packages/ts-transform/src/utils/css-transform.tsx:54]: done with the shorthand expressions i see 😝

File: packages/ts-transform/src/utils/string-interpolations.tsx [hunk 2]
```diff
@@ -39,7 +44,12 @@ export const cssAfterInterpolation = (css: string): { css: string; variableSuffi
   };
 };
 
-export const cssBeforeInterpolation = (css: string): { css: string; variablePrefix?: string } => {
+export interface BeforeInterpolation {
+  css: string;
+  variablePrefix?: string;
+}
```
Comment (danieldelcore) [inline packages/ts-transform/src/utils/string-interpolations.tsx:50]: maybe merge these two definitions?

`interface Interpolation` ?

File: packages/ts-transform/src/utils/css-transform.tsx (hunk 1 above)
Comment (madou) [reply packages/ts-transform/src/utils/css-transform.tsx:54]: oh i think prettier did it

File: packages/ts-transform/src/utils/template-literal-to-css.tsx [hunk 3]
```diff
@@ -190,6 +183,7 @@ export const templateLiteralToCss = (
         const result = evaluateFunction(declarationNode, collectedDeclarations, context);
         css += result.css;
         cssVariables = cssVariables.concat(result.cssVariables);
+        // TODO: DOUBLE CHECK THIS NEEDS PREFIX EXTRACTION ^
```
Comment (danieldelcore) [inline packages/ts-transform/src/utils/template-literal-to-css.tsx:186]: Are these intentional?

Comment (madou) [reply packages/ts-transform/src/utils/template-literal-to-css.tsx:186]: yeah

**Pull request merged**